### PR TITLE
Importer Tests (#932)

### PR DESCRIPTION
**Pull request by He3lixxx**
Added all the tests mentioned in #932

### CODE REVIEW

File: evap/staff/tests/test_views.py [hunk 1]
```diff
@@ -296,6 +382,8 @@ def test_import_valid_file(self):
         form["excel_file"] = (self.filename_valid,)
         page = form.submit(name="operation", value="test")
 
+        self.assertEqual(original_user_count, original_user_count)
```
Comment (karyon) [inline evap/staff/tests/test_views.py]: check that again plz

File: evap/staff/tests/test_views.py [hunk 2]
```diff
@@ -322,16 +410,18 @@ def test_import_invalid_file(self):
...
         form = page.forms["semester-import-form"]
-        form["excel_file"] = (self.filename_valid,)
-        form.submit(name="operation", value="test")
+        form["excel_file"] = (self.filename_random,)
+        reply = form.submit(name="operation", value="test")
 
-        self.assertEqual(self.course.participants.count(), original_participant_count)
+        self.assertEqual(UserProfile.objects.count(), original_user_count)
+        self.assertContains(reply, "Couldn&#39;t read the file. Error: Unsupported format, or corrupt file: Expected BOF record;")
+        self.assertNotContains(reply, 'Import previously uploaded file')
```
Comment (karyon) [inline evap/staff/tests/test_views.py]: to me this looks like it tests the importer, not the view.

File: evap/staff/tests/test_views.py [hunk 3]
```diff
@@ -136,6 +136,92 @@ def test_deletes_users(self):
...
+    def test_import_wrong_file(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["user-import-form"]
+        form["excel_file"] = (self.filename_random,)
+
+        reply = form.submit(name="operation", value="test")
+
+        self.assertContains(reply, "Couldn&#39;t read the file. Error: Unsupported format, or corrupt file: Expected BOF record;")
+        self.assertNotContains(reply, 'Import previously uploaded file')
```
Comment (karyon) [inline evap/staff/tests/test_views.py]: to me this looks like it tests the importer, not the view.

File: evap/staff/tests/test_views.py [hunk 4]
```diff
@@ -734,6 +905,40 @@ def test_suspicious_operation(self):
...
+        form = page.forms["contributor-import-form"]
+        reply = form.submit(name="operation", value="import-contributors", expect_errors=True)
+
+        self.assertEqual(reply.status_code, 400)
+
+    def test_invalid_participant_import_operation(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["participant-import-form"]
+        reply = form.submit(name="operation", value="import-participants", expect_errors=True)
```
Comment (karyon) [inline evap/staff/tests/test_views.py]: again, what's invalid about this and the above operation?

File: evap/staff/tests/test_views.py (hunk 2 above)
Comment (karyon) [inline evap/staff/tests/test_views.py]: OTOH, having one test that checks whether errors from the importer are properly handled by the view are good to have, too.

File: evap/staff/tests/test_views.py [hunk 5]
```diff
@@ -136,6 +136,92 @@ def test_deletes_users(self):
...
+        page = form.submit(name="operation", value="test")
+
+        self.assertContains(page, 'Please select an Excel file')
+        self.assertNotContains(page, 'Import previously uploaded file')
+
+    def test_invalid_import_operation(self):
+        page = self.app.get(self.url, user='staff')
+
+        form = page.forms["user-import-form"]
+        reply = form.submit(name="operation", value="import", expect_errors=True)
```
Comment (He3lixxx) [inline evap/staff/tests/test_views.py]: No excel file was uploaded and tested previously. This POST request could only happen with malicious intention.

File: evap/staff/tests/test_views.py [hunk 6]
```diff
@@ -854,12 +855,27 @@ def test_import_invalid_participants_file(self):
...
+        form["excel_file"] = (self.filename_valid,)
+
+        reply = form.submit(name="operation", value="test-participants")
+        self.assertContains(reply, "The existing user would be overwritten with the following data:<br>"
+                " - lucilia.manilium ( None None, 42@42.de) (existing)<br>"
+                " - lucilia.manilium ( Lucilia Manilium, lucilia.manilium@institution.example.com) (new)")
+
+        # delete the uploaded file again so other tests can start with no file guaranteed
+        form = reply.forms["participant-import-form"]
+        form.submit(name="operation", value="import-participants")
```
Comment (karyon) [inline evap/staff/tests/test_views.py]: please check whether you can write code that explicitly deletes the file.

File: evap/staff/tests/test_views.py [hunk 7]
```diff
@@ -870,29 +886,22 @@ def test_import_invalid_contributors_file(self):
         self.assertContains(reply, 'Errors occurred while parsing the input data. No data was imported.')
         self.assertNotContains(reply, 'Import previously uploaded file')
 
-        self.assertEqual(UserProfile.objects.count(), original_user_count)
+    def test_import_contributors_warning_handling(self):
+        # Tests whether warnings given from the importer are displayed
```
Comment (karyon) [inline evap/staff/tests/test_views.py]: please use `""" comment goes here """` style for comments describing the whole function.

**Pull request merged**